common/src/service/IotHubConnectionString.cs

```diff
@@ -31,19 +30,8 @@ public IotHubConnectionString(IotHubConnectionStringBuilder builder)
             IotHubName = builder.IotHubName;
             HttpsEndpoint = new UriBuilder("https", HostName).Uri;
             AmqpEndpoint = new UriBuilder(CommonConstants.AmqpsScheme, builder.HostName, AmqpConstants.DefaultSecurePort).Uri;
-            DeviceId = builder.DeviceId;
```

Comment:
This was copied over from DeviceClient and is not needed in ServiceClient. We don't make use of it anyway. I am running tests to ensure nothing breaks.